osu.Server.Spectator/Hubs/CountdownImplementation.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using osu.Game.Online.Multiplayer;
+using osu.Game.Online.Multiplayer.Countdown;
+
+namespace osu.Server.Spectator.Hubs
+{
+    /// <summary>
+    /// The implementation for a countdown timer.
+    /// </summary>
+    public class CountdownImplementation
```

Comment:
I find it a touch weird that this is an `Implementation`, not a `ServerCountdown : Countdown` in a similar fashion to how `ServerMultiplayerRoom` is structured. Is there reasoning behind this? ie. are there planned to be different types of countdowns in the future?

Reply:
Also, because it was mentioned in a dupe comment and I missed it, I don't really see how `Implementation` necessitates the need for "more than one implementation", and if you're going to bring up `TeamVersus` vs `MatchTypeImplementation` (base class) then I'd mention that I struggle with the naming and would prefer `TeamVersusImplementation`.